packages/astro/src/core/build/generate.ts

```diff
@@ -145,10 +146,15 @@ export async function generatePages(opts: StaticBuildOptions, internals: BuildIn
 		const baseDirectory = getOutputDirectory(opts.settings.config);
 		const renderersEntryUrl = new URL('renderers.mjs', baseDirectory);
 		const renderers = await import(renderersEntryUrl.toString());
+		let middleware: MiddlewareHandler = (_, next) => next();
+		try {
+			middleware = await import(new URL('middleware.mjs', baseDirectory).toString()).then(mod => mod.onRequest);
+		} catch {}
```

Comment:
Let's add a comment that says why we use a `try/catch` and why we don't check the error. Context is important